lib/remoteproc/remoteproc_virtio.c

```diff
@@ -40,6 +40,9 @@ static unsigned char rproc_virtio_get_status(struct virtio_device *vdev)
 	rpvdev = metal_container_of(vdev, struct remoteproc_virtio, vdev);
 	vdev_rsc = rpvdev->vdev_rsc;
 	io = rpvdev->vdev_rsc_io;
+
+	RSC_TABLE_INVALIDATE(vdev_rsc, sizeof(struct fw_rsc_vdev));
+
```

Comment:
This is fine now, thanks.

Shouldn't we also `RSC_TABLE_FLUSH` in `rproc_virtio_set_status()` though?

Reply:
@arnopo I'm not familiar with remoteproc at all. What are the structs that are actually r/w from both sides that needs to be flushed / invalidated?

Reply:
@iuliana-prodan, @carlocaione 

Yes we need cache operation even in read config and write config as well.
One more question, should we perform only flush or invalidate and flush both during writing resource table? Or it is really platform dependent?

Reply:
"Flush" and "invalidate" should be sufficient if the structures are aligned on a cache line. We can improve this in the future if needed.

Reply:
Right @TanmayShah-xilinx, 
The cache shoule also be managed in `rproc_virtio_read_config` and`rproc_virtio_read_config`